ui/non_interactive.go

```diff
@@ -40,12 +42,21 @@ func (ui *NonInteractiveUI) PrintTable(table Table) {
 	ui.parent.PrintTable(table)
 }
 
-func (ui *NonInteractiveUI) AskForText(label string) (string, error) {
-	panic("Cannot ask for input in non-interactive UI")
+func (ui *NonInteractiveUI) AskForText(opts TextOpts) (string, error) {
+	if opts.ValidateFunc != nil {
+		isValid, message, err := opts.ValidateFunc(opts.Default)
+		if err != nil || !isValid {
+			return "", fmt.Errorf("Validation Error: %s", message)
+		}
+	}
+	return opts.Default, nil
 }
 
-func (ui *NonInteractiveUI) AskForChoice(label string, options []string) (int, error) {
-	panic("Cannot ask for a choice in non-interactive UI")
+func (ui *NonInteractiveUI) AskForChoice(opts ChoiceOpts) (int, error) {
+	if opts.Default >= len(opts.Choices) || opts.Default < 0 {
+		return 0, fmt.Errorf("Default value should be index and must be in (0-%d)\n", len(opts.Choices)-1)
```

Comment:
remove newline in error message (we only use them if error message is multi-line)

Reply:
Got it. Adopted.